### PR TITLE
[CSL-1193] Get rid of explorer code in cardano-sl lib and with-explorer flag

### PR DESCRIPTION
**Pull request by pva701**

### CODE REVIEW

**Comment by gromakovsky** [comment]
There are 6 conflicts already, btw.

File: lib/src/Pos/Generator/BlockEvent.hs [hunk 1]
```diff
@@ -200,10 +204,11 @@ genBlocksInStructure ::
        , Functor t, Foldable t)
     => AllSecrets
     -> GenesisWStakeholders
+    -> TxpGlobalSettings
     -> Map Path BlockDesc
     -> t Path
     -> RandT g m (t BlundDefault)
-genBlocksInStructure secrets bootStakeholders annotations s = do
+genBlocksInStructure secrets bootStakeholders txpSettings annotations s = do
```
Comment (gromakovsky) [inline lib/src/Pos/Generator/BlockEvent.hs]: Why did you add it as arguments everywhere in this module instead of putting into reader context?

File: lib/src/Pos/Generator/BlockEvent/DSL.hs [hunk 2]
```diff
@@ -119,11 +120,12 @@ runBlockEventGenT
     :: BlockTxpGenMode g ctx m
     => AllSecrets
     -> GenesisWStakeholders
+    -> TxpGlobalSettings
     -> BlockEventGenT g m ()
     -> RandT g m BlockScenario
-runBlockEventGenT secrets genStakeholders m = do
+runBlockEventGenT secrets genStakeholders txpSettings m = do
```
Comment (gromakovsky) [inline lib/src/Pos/Generator/BlockEvent/DSL.hs]: Why not add to context instead?

File: lib/src/Pos/Generator/BlockEvent.hs (hunk 1 above)
Comment (pva701) [reply lib/src/Pos/Generator/BlockEvent.hs]: makes sense

**Comment by pva701** [comment]
@gromakovsky I've fixed all issues and force pushed, can you approve if everything is ok?

**Comment by gromakovsky** [comment]
> The Travis CI build failed 

:loh:
